### PR TITLE
To add Cosmic's Additional Locations to the game

### PR DESCRIPTION
**Pull request by cosmiccoincidence**
This request is to add my (CosmicCoincidence) additional location mods to the base game (not as official mods but part of the vanilla game). These additional locations include...

|LOCATION TYPE|LOCATION MAP SIZE|NUMBER OF VARIATIONS|LOCATION SPAWN CATEGORY|DESCRIPTION|
|:-|:-|:-|:-|:-|
|Cabin|1X1|5|Special|Additional to the in-game cabins.|
|Desolate Barn|1X1|10|Special|Forgotten and falling apart barns that spawn in fields/woods.|
|Empty Commercial Lot|1X1|10|Shop|Empty lot where a store could be built.|
|Empty Residential Lot|1X1|5|House|Empty lot where a house could be built.|
|Abandoned Warehouse|1X1|5|Shop|Abandoned warehouse which may have some junk left behind.|
|Vacant House|1X1|20|House|Houses in various states of condition (construction, repair, moving, garage sale, etc.).|
|Dump|1X1|5|Shop|City dump filled with piles of rubble and trash (you may find something of use in here, but its probably mostly trash)|
|Dumpsite|1X1|10|Special|Illegal dump sites found in fields or forests. Where do people leave there old washing machines? In the woods of course!|
|Landfill|1X1|5|Shop|City landfill, similar to dumps.|
|Recycle Center|1X1|3|Shop|Center meant for sorting junk into it's base materials, steel, glass, plastic, electronics, etc. before being recycled.|
|Junkyard|2X2|1|Special|Large yard of wasted vehicles with a garage. BEWARE of dogs!|
|Cemetery|1X1|5|Park|Additional to the in-game cemeteries.|
|Cemetery|1X2|1|Special|A 'medium' cemetery, larger than the standard (small) cemeteries but smaller than the religious cemeteries.|
|Community Garden|1X1|3|Park|A public use garden for growing produce.|
|Public Garden|1X1|2|Park|A public garden filled with flowers and trees. A beautiful spot to stop and read a book on a nice day.|
|Botanical Garden|1X2|1|Special|A larger garden with a greenhouse, professionally displaying various botanical species.|
|Park|1X1|5|Park|Additional to the in-game parks.|
|Public Pond|1X2|1|Special|A larger park/pond combination.|
|Nature Trail|1X2|1|Special|A larger wooded trail.|
|Tree Farm|1X2|1|Special|A farm for pine trees. "Were only open in December"|
|Pool|1X1|5|Park|Additional to the in-game pools. 2 are outdoor, 3 are indoor.|
|Campground|2X2|1|Special|A large wooded area with multiple campsites and community bath house.|
|Shooting Range|1X2|1|Special|"Come on down to the Dickerson's Family Shooting Range for some fun! We got targets up to 30 yards!"|
|Landscaping Supply Co|1X2|1|Special|Shop with a lot full of piles dirt, sand, rocks, cement, wood, etc.|
|LAN Center|1X1|2|Shop|A place where kids without their own computers go to play video games and use the internet.|
|Dollar Store|1X1|2|Shop|Cheap goods of various categories all in one store.|

### CODE REVIEW

**Comment by John-Candlebury** [comment]
I'm not sold on the empty lot terrains. They take city space that could be used by interesting and usefull buildings and replace them with what is basically nothing, 

Our cities already generate with what are tentatively empty lots (AKA mid city empty fields)Maybe consider editing mapgen code so that those fields become empty lots?

**Comment by cosmiccoincidence** [comment]
I don't know how to directly reply but responding to the empty lots. I definitely hear and understand that argument, I tried to offset that from being too much of an issue be keeping most of spawn rates for those kinda things low. Also the empty lots although not as interesting as most places, can spawn trash, deserted vehicles, food vendor carts, not all are just an empty field.

Also I just linted ALL my files as well as threw in a couple other buildings... Landscaping Supply Co, LAN Center, and Dollar Store

**Comment by ZhilkinSerg** [comment]
Jenkins rebuild.

File: data/json/mapgen/cosmics_additions/DollarStore.json [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+        ".": "t_floor",
+        "_": "t_grass",
+        "d": "t_dirt",
+        "s": "t_sidewalk",
+        "{": "t_floor",
+        "|": "t_wall"
+      }
+    }
+  }
+]
```
Comment (ZhilkinSerg) [inline data/json/mapgen/cosmics_additions/DollarStore.json]: You have EOL missing in the end of several files, so JSON formatting fails.

File: data/json/mapgen/cosmics_additions/LANCenter.json [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+        "v": "t_floor",
+        "{": "t_floor",
+        "|": "t_brick_wall"
+      }
+    },
+    "om_terrain": "lancenter",
+    "type": "mapgen",
+    "weight": 100
+  }
+]
```
Comment (ZhilkinSerg) [inline data/json/mapgen/cosmics_additions/LANCenter.json]: EOL missing.

Comment (cosmiccoincidence) [reply data/json/mapgen/cosmics_additions/LANCenter.json]: EOL?

Comment (Zireael07) [reply data/json/mapgen/cosmics_additions/LANCenter.json]: End of line mark, usually inserted by your text editor. What do you use?

Comment (cosmiccoincidence) [reply data/json/mapgen/cosmics_additions/LANCenter.json]: Notepad++

Comment (Nymall) [reply data/json/mapgen/cosmics_additions/LANCenter.json]: Just put a extra row at the end of the JSON and it should fix the issue

Comment (cosmiccoincidence) [reply data/json/mapgen/cosmics_additions/LANCenter.json]: thanks, changes mades

**Comment by ZhilkinSerg** [comment]
Mechanical winch in abandoned warehouse is not working and it is possible to open doors separately.

image

**Comment by cosmiccoincidence** [comment]
Warehouse garage doors fixed.

**Comment by kevingranade** [comment]
This pull request has been mentioned on **Cataclysm: Dark Days Ahead**. There might be relevant details there:

https://discourse.cataclysmdda.org/t/potential-engine-solutions-and-fun-stuff/17904/13

**Pull request merged**